compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -3229,7 +3229,7 @@ class TrackingTypeComparer(initctx: Context) extends TypeComparer(initctx) {
             MatchTypeTrace.noInstance(scrut, cas, fails)
             NoType
           case MatchResult.Reduced(tp) =>
-            tp
+            tp.simplified
```

Comment:
No, we should never use `simplified` in TypeComparer. `tryNormalize` is OK, simplified is not

Reply:
There's simplified everywhere in match type reduction: in `paramInstances` and in one of the two types of reduction.  All I'm doing is making the two reduction cases the same.  That way `case (0,T) => EmptyTuple` isn't treated differently from `case (0,_) => EmptyTuple`.

Btw, in case you missed it, we addressed an issue with the use of simplified in match type reduction in #17937 by making it run under Mode.Type, to avoid how it interacts with erasure.

This one change makes reduction return a simpler type, which solves a bunch of compilation issues in match type.  It also solves a bunch of problems in test-pickler - I finally didn't have to add these test cases to the exclude list, and I would expect some more can be removed.

Reply:
Agreed that the two should behave the same. But why do we need `simplfy` (as opposed to `tryNormalize`? It was not made for this use case. The idea of `simplify` was to simplify inferred types so that they look simpler and that type inference is helped.

`simplify` can force type variables, or map type parameters to type variables, or re-arrange unions and intersections. None of this should happen at the very low level where we reduce match types. 

In short `simplify` is a method that makes sense for type _inference_, not for basic type computations.

Reply:
Normalize doesn't traverse all applied types, like `(3 - 1, Int)`.

Reply:
@odersky shall I create a less powerful simplify and use that within match type reduction, or should I make all applied types try to normalize their components in tryNormalize?

Reply:
Why do we have to do it deeply at all? Can't we normalize when we inspect the element?

Reply:
The `summon[Nat.FromInt[3]]` is one example which fails by not just simplifying the reductions.  But there are other examples where normalization isn't called: in Type#atoms, in TypeComparer#provablyDisjoint.